pkg/runtime/processor/pubsub/streamerconn.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub
+
+import (
+	"context"
+	"sync"
+
+	rtv1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+)
+
+type streamconn struct {
+	lock             sync.RWMutex
+	s                *streamer
+	publishResponses map[string]chan *rtv1pb.SubscribeTopicEventsResponseAlpha1
+	stream           rtv1pb.Dapr_SubscribeTopicEventsAlpha1Server
+}
+
+func (s *streamconn) notifyPublishResponse(ctx context.Context, resp *rtv1pb.SubscribeTopicEventsResponseAlpha1) {
+	s.lock.RLock()
+	ch, ok := s.publishResponses[resp.GetId()]
+	s.lock.RUnlock()
+
+	if !ok {
+		log.Errorf("no client stream expecting publish response for id %q", resp.GetId())
```

Comment:
Is there a scenario where this is a valid path and thus warrants a Warn instead of Error?

Reply:
No, this likely means that the client is reporting the processing result of a message with an ID which has already been reported or was never published to the app in the first place.